### PR TITLE
resource release bug fixed

### PR DESCRIPTION
**Pull request by hzcyf**
1.  resource release bug fixed; 
2. change the default clock sync modeto sync_with_host ;
3. Enhance code robustness；
4. add k4a_enable_soft_filter interface；
5. add k4a_get_wired_sync_mode interface;
6. switch OrbbecSDK version to v1.9.5;
7. update document;
8. update version to 1.9.2.

### CODE REVIEW

**Review by zhonghong322 — COMMENTED** [review]
1、假设使用清0功能，如果先开imu，然后在开启视频流，这时清0命令会发送2次吗？

**Review by zzuliys — APPROVED** [review]

File: include/k4a/k4a.h [hunk 1]
```diff
@@ -33,18 +33,54 @@ extern "C" {
...
+/** set device soft filter switch
  *
  * \param device_handle
  * Output parameter which on success will return a handle to the device.
  *
- * \param param
- * K4A_DEVICE_CLOCK_SYNC_MODE_RESET: The delay time of executing the timestamp reset function after receiving the command or signal in microseconds.
- * K4A_DEVICE_CLOCK_SYNC_MODE_SYNC:  The interval for auto-repeated synchronization, in microseconds. If the value is 0, synchronization is performed only once.
+ * \param filter_switch
+ * Device clock synchronization mode
```
Comment (hzcyf) [inline include/k4a/k4a.h:58]: This comment is wrong.

File: src/orbbec/ob_k4a_impl.cpp [hunk 2]
```diff
@@ -569,31 +595,50 @@ k4a_result_t init_device_context(k4a_device_t device_handle)
...
-    while(device_ctx->clock_sync_cv.wait_for(lck,std::chrono::microseconds(intervalMicroseconds)) == std::cv_status::timeout){
+    while(device_ctx->clock_sync_cv.wait_for(lck,std::chrono::microseconds(interval_usec)) == std::cv_status::timeout){
         if(device_ctx->current_device_clock_sync_mode == K4A_DEVICE_CLOCK_SYNC_MODE_SYNC){
             if(device_ctx->device){
                 ob_device_timer_sync_with_host(device_ctx->device, &ob_err);
             }
         }
     }
 }
+k4a_result_t k4a_set_soft_filter(k4a_device_t device_handle, bool filter_switch){
```
Comment (hzcyf) [inline src/orbbec/ob_k4a_impl.cpp:617]: Perhaps `k4a_result_t k4a_enable_soft_filter(k4a_device_t device_handle, bool enable)` could be a better option.

File: include/k4a/k4a.hpp [hunk 3]
```diff
@@ -1540,8 +1540,8 @@ class device
         k4a_set_soft_filter(m_handle, filter_switch);
     }
 
-    void get_device_sync_mode(){
-        k4a_get_device_sync_mode(m_handle);
+    k4a_wired_sync_mode_t get_device_sync_mode(){
```
Comment (hzcyf) [inline include/k4a/k4a.hpp:1543]: Changing the function name to `get_wired_sync_mode` might better fit the original concept.

**Pull request merged**